ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -123,19 +126,37 @@ public void onPartitionRemoved(PartitionId partitionId) {
 
     // start background persistent thread
     // start scheduler thread to persist index in the background
-    scheduler.scheduleAtFixedRate(persistor, replicationConfig.replicationTokenFlushDelaySeconds,
-        replicationConfig.replicationTokenFlushIntervalSeconds, TimeUnit.SECONDS);
+    scheduleTask(persistor, true, replicationConfig.replicationTokenFlushDelaySeconds,
+        replicationConfig.replicationTokenFlushIntervalSeconds, "replica token persistor");
 
-    if (cloudConfig.cloudBlobCompactionEnabled) {
-      // Schedule thread to purge dead blobs for this VCR's partitions
-      // after delay to allow startup to finish.
-      long delaySec = cloudConfig.cloudBlobCompactionStartupDelaySecs;
-      long intervalSec = TimeUnit.HOURS.toSeconds(cloudConfig.cloudBlobCompactionIntervalHours);
-      scheduler.scheduleAtFixedRate(cloudStorageCompactor, delaySec, intervalSec, TimeUnit.SECONDS);
-      logger.info("Scheduled compaction task to run every {} hours starting in {} seconds.",
-          cloudConfig.cloudBlobCompactionIntervalHours, delaySec);
+    // Schedule thread to purge dead blobs for this VCR's partitions
+    // after delay to allow startup to finish.
+    scheduleTask(cloudStorageCompactor, cloudConfig.cloudBlobCompactionEnabled,
+        cloudConfig.cloudBlobCompactionStartupDelaySecs,
+        TimeUnit.HOURS.toSeconds(cloudConfig.cloudBlobCompactionIntervalHours), "cloud blob compaction");
+
+    // Schedule thread to purge blobs belonging to deprecated containers for this VCR's partitions
+    // after delay to allow startup to finish.
+    scheduleTask(() -> cloudContainerCompactor.compactAssignedDeprecatedContainers(
+        virtualReplicatorCluster.getAssignedPartitionIds()), cloudConfig.cloudContainerCompactionEnabled,
+        cloudConfig.cloudContainerCompactionStartupDelaySecs,
+        TimeUnit.HOURS.toSeconds(cloudConfig.cloudContainerCompactionIntervalHours), "cloud container compaction");
+  }
```

Comment:
According to your design, there would be several threads concurrently performing container compaction. Could you point me to the implementation?